user_tools/src/spark_rapids_pytools/common/sys_storage.py

```diff
@@ -375,3 +376,117 @@ def is_file_path(self, value: str):
+            return FSUtil.download_from_url(file_url + '.asc', dest_folder)
+        except urllib.error.URLError:
+            return None
+
+    @classmethod
+    def get_integrity_algorithm(cls, hash_info: dict):
+        for algorithm in cls.SUPPORTED_ALGORITHMS:
+            if algorithm in hash_info:
+                return algorithm
+        return None
+
+    @classmethod
+    def _check_integrity_using_gpg(cls, file_path: str, signature_file_path: str) -> bool:
+        """
+        Verify file integrity using GPG and its corresponding .asc signature file.
+        Note - The verification has a timeout of `GPG_TIMEOUT_SEC`
+
+        :param file_path: Path to the file to be verified.
+        :param signature_file_path: Path to the .asc signature file.
+        :return: True if the file integrity is valid and the signature is verified, False otherwise.
+        """
+        if not (os.path.isfile(file_path) and os.path.isfile(signature_file_path)):
+            return False
+
+        if which('gpg') is None:
+            # GPG is not found in the system
+            return False
+
+        gpg_command = [
+            'gpg',
+            '--auto-key-locate keyserver',
+            '--keyserver pgp.mit.edu',
+            '--keyserver-options auto-key-retrieve',
+            '--verify',
+            signature_file_path,
+            file_path
+        ]
+
+        try:
+            result = subprocess.check_output(' '.join(gpg_command),
+                                             shell=True,
+                                             text=True,
+                                             timeout=cls.GPG_TIMEOUT_SEC,
+                                             stderr=subprocess.STDOUT)
+            return 'Good signature' in result
+        except (subprocess.TimeoutExpired, subprocess.CalledProcessError):
+            return False
+
+    @classmethod
+    def _check_integrity_using_algorithm(cls, file_path: str, algorithm: str, expected_hash: str) -> bool:
+        """
+        Checks the integrity of a downloaded file by calculating its hash and comparing it with the expected hash.
+
+        :param file_path: Path of the downloaded file.
+        :param algorithm: Name of the hash algorithm to use for calculating
+        :param expected_hash: Expected hash value for the file
+        :return: True if the calculated hash matches the expected hash, False otherwise.
+        """
+        if not os.path.isfile(file_path):
+            # Cannot verify file
+            return False
+
+        # Helper function to calculate the hash of the file using the specified algorithm
+        def calculate_hash(hash_algorithm='sha256'):
+            if hash_algorithm not in cls.SUPPORTED_ALGORITHMS:
+                # Unsupported hash algorithm
+                return False
+            hash_function = cls.SUPPORTED_ALGORITHMS[hash_algorithm]()
+            with open(file_path, 'rb') as file:
+                while chunk := file.read(8192):
```

Comment:
Number 8192 can be made constant variable and referenced here.

Reply:
Replaced by a constant variable